### ISSUE TITLE
if disconnected, does not try to reconnect, does not exit

### ISSUE THREAD

**Issue by ghostsquad** [xBytez/slackbotapi#23]
I encountered a problem where the bot was disconnected from the internet, and the app didn't crash, nor try to connect again.

**Comment by franciskim** [comment]
I have this same issue - I don't think it's a bug but an improvement.

**Comment by ghostsquad** [comment]
If I were to hook up monitoring for the app, it would never indicate there was a problem, therefore I would consider this a bug. I would expect at the very least for an exception or some other indicator that I can watch for and either attempt to reconnect on my own, or log and exit. I think an improvement here would be to auto-reconnect with retry and wait/timeout options (default: false, 0).

**Comment by franciskim** [comment]
the below triggers for me, it doesn't for you?
```
self.out('warning', 'Disconnected. Error: '+data);
self.emit("close", data);
```

**Comment by xBytez** [comment]
Fixed by @franciskim in pull #33 ! Great job, thanks so much.

**Comment by franciskim** [comment]
@xBytez all good!!
@ghostsquad are you able to test the new code in your environment? I'm not seeing disconnections after the code update :)

**Comment by xBytez** [comment]
Bug fixed in #36 & #33, seems to work fine. Closing for now.
